gear-programs/vara-tokenizer/tests/gtest.rs

```diff
@@ -0,0 +1,175 @@
+
+    let total_supply = client
+        .total_supply()
+        .recv(program_id)
+        .await
+        .expect("Failed");
+
+    // assert
+    assert!(total_supply.is_zero());
+}
+
+#[tokio::test]
+async fn mint_from_value_works() {
+    let (remoting, program_code_id) = init_remoting();
+
+    let program_factory = vara_tokenizer_client::VaraTokenizerFactory::new(remoting.clone());
+
+    let program_id = program_factory
+        .new("Name".into(), "Symbol".into(), 10u8)
+        .send_recv(program_code_id, b"salt")
+        .await
+        .unwrap();
+
+    let initial_balance = remoting.system().balance_of(ADMIN_ID);
+    let mint_value = 10_000_000_000_000;
+
+    let mut client = vara_tokenizer_client::Tokenizer::new(remoting.clone());
+
+    client
+        .mint_from_value()
+        .with_value(mint_value)
+        .send_recv(program_id)
+        .await
+        .expect("Failed send_recv")
+        .expect("Failed to mint from value");
+
+    let balance = remoting.system().balance_of(ADMIN_ID);
+    let program_balance = remoting.system().balance_of(program_id);
+    // TODO update test after next `gtest` release
+    assert_eq!(balance, initial_balance - mint_value);
+    assert_eq!(program_balance, mint_value);
+}
+
+#[tokio::test]
+async fn burn_and_return_value_works() {
+    let (remoting, program_code_id) = init_remoting();
+
+    let program_factory = vara_tokenizer_client::VaraTokenizerFactory::new(remoting.clone());
+
+    let program_id = program_factory
+        .new("Name".into(), "Symbol".into(), 10u8)
+        .send_recv(program_code_id, b"salt")
+        .await
+        .unwrap();
+
+    let initial_balance = remoting.system().balance_of(ADMIN_ID);
+    let mint_value = 10_000_000_000_000;
+
+    let mut client = vara_tokenizer_client::Tokenizer::new(remoting.clone());
+
+    client
+        .mint_from_value()
+        .with_value(mint_value)
+        .send_recv(program_id)
+        .await
+        .expect("Failed send_recv")
+        .expect("Failed to mint from value");
+
+    let client_balance = client
+        .balance_of(ADMIN_ID.into())
+        .recv(program_id)
+        .await
+        .unwrap();
+
+    let balance = remoting.system().balance_of(ADMIN_ID);
+    let program_balance = remoting.system().balance_of(program_id);
+    assert_eq!(balance, initial_balance - mint_value);
+    assert_eq!(program_balance, mint_value);
+    assert_eq!(client_balance, mint_value.into());
+
+    client
+        .burn_and_return_value(mint_value)
+        .send_recv(program_id)
+        .await
+        .expect("Failed send_recv")
+        .expect("Failed to burn and return value");
+
+    let client_balance = client
+        .balance_of(ADMIN_ID.into())
+        .recv(program_id)
+        .await
+        .unwrap();
+
+    let balance = remoting.system().balance_of(ADMIN_ID);
+    let program_balance = remoting.system().balance_of(program_id);
+    // TODO update test after next `gtest` release
```

Comment:
Please describe what changes you expect in new release or link an issue, if possible